Source/WebGPU/WebGPU/Buffer.mm

```diff
@@ -95,33 +95,41 @@ static bool validateCreateBuffer(const Device& device, const WGPUBufferDescripto
     return true;
 }
 
-static MTLStorageMode storageMode(bool deviceHasUnifiedMemory, WGPUBufferUsageFlags usage, bool mappedAtCreation)
-{
-    if (deviceHasUnifiedMemory)
-        return MTLStorageModeShared;
-#if PLATFORM(MAC) || PLATFORM(MACCATALYST)
```

Comment:
only on Intel Macs did we ever execute lines 102..111

Using Managed instead of Private is fine with some potential minimal performance impact to AMD, but we rarely allocated a private buffer anyways .